### PR TITLE
Add a test to verify that command metadata conforms to expectations.

### PR DESCRIPTION
**Pull request by timburks**
[[ Reviewers FYI - I don't think this needs everyone's detailed review (though comments are welcome!), but I want to let you know that this adds automatic checks of certain style characteristics of the registry tool command info. Like golangci-lint, don't worry about knowing the rules, just notice when the test fails. ]]

This directly addresses #909.

The first commit in this PR adds a test that checks metadata on the Cobra commands that are declared in the `registry` tool. Initially all problems are logged, but in subsequent commits, the logs will be changed to errors and associated problems will be fixed. Here's the initial test output:
```
$ go test ./cmd/registry/cmd  -v
=== RUN   TestCommandMetadata
=== RUN   TestCommandMetadata/registry
=== RUN   TestCommandMetadata/registry_annotate
    root_test.go:73: "registry annotate" "filter" flag description must not be empty.
    root_test.go:73: "registry annotate" "jobs" flag description must not be empty.
    root_test.go:73: "registry annotate" "overwrite" flag description must not be empty.
=== RUN   TestCommandMetadata/registry_apply
    root_test.go:73: "registry apply" "parent" flag description must not be empty.
    root_test.go:77: "registry apply" "recursive" flag description must not begin with an upper case letter.
=== RUN   TestCommandMetadata/registry_auth
    root_test.go:39: "registry auth" short description must not end with a period.
    root_test.go:46: "registry auth" short description must refer to `the API Registry` instead of `the registry`.
=== RUN   TestCommandMetadata/registry_auth_print-token
    root_test.go:39: "registry auth print-token" short description must not end with a period.
=== RUN   TestCommandMetadata/registry_check
    root_test.go:46: "registry check" short description must refer to `the API Registry` instead of `the registry`.
=== RUN   TestCommandMetadata/registry_compute
=== RUN   TestCommandMetadata/registry_compute_complexity
=== RUN   TestCommandMetadata/registry_compute_conformance
=== RUN   TestCommandMetadata/registry_compute_lint
    root_test.go:73: "registry compute lint" "linter" flag description must not be empty.
=== RUN   TestCommandMetadata/registry_compute_lintstats
    root_test.go:73: "registry compute lintstats" "linter" flag description must not be empty.
=== RUN   TestCommandMetadata/registry_compute_score
=== RUN   TestCommandMetadata/registry_compute_scorecard
=== RUN   TestCommandMetadata/registry_compute_vocabulary
=== RUN   TestCommandMetadata/registry_config
    root_test.go:39: "registry config" short description must not end with a period.
=== RUN   TestCommandMetadata/registry_config_configurations
=== RUN   TestCommandMetadata/registry_config_configurations_activate
    root_test.go:39: "registry config configurations activate" short description must not end with a period.
=== RUN   TestCommandMetadata/registry_config_configurations_create
    root_test.go:39: "registry config configurations create" short description must not end with a period.
=== RUN   TestCommandMetadata/registry_config_configurations_delete
    root_test.go:39: "registry config configurations delete" short description must not end with a period.
=== RUN   TestCommandMetadata/registry_config_configurations_describe
    root_test.go:39: "registry config configurations describe" short description must not end with a period.
=== RUN   TestCommandMetadata/registry_config_configurations_list
    root_test.go:39: "registry config configurations list" short description must not end with a period.
=== RUN   TestCommandMetadata/registry_config_get
    root_test.go:39: "registry config get" short description must not end with a period.
=== RUN   TestCommandMetadata/registry_config_list
    root_test.go:39: "registry config list" short description must not end with a period.
=== RUN   TestCommandMetadata/registry_config_set
    root_test.go:39: "registry config set" short description must not end with a period.
=== RUN   TestCommandMetadata/registry_config_unset
    root_test.go:39: "registry config unset" short description must not end with a period.
=== RUN   TestCommandMetadata/registry_count
=== RUN   TestCommandMetadata/registry_count_revisions
    root_test.go:73: "registry count revisions" "filter" flag description must not be empty.
=== RUN   TestCommandMetadata/registry_count_versions
    root_test.go:73: "registry count versions" "filter" flag description must not be empty.
=== RUN   TestCommandMetadata/registry_delete
    root_test.go:73: "registry delete" "filter" flag description must not be empty.
    root_test.go:73: "registry delete" "jobs" flag description must not be empty.
=== RUN   TestCommandMetadata/registry_diff
    root_test.go:46: "registry diff" short description must refer to `the API Registry` instead of `the registry`.
=== RUN   TestCommandMetadata/registry_export
=== RUN   TestCommandMetadata/registry_export_csv
    root_test.go:73: "registry export csv" "filter" flag description must not be empty.
=== RUN   TestCommandMetadata/registry_export_yaml
    root_test.go:46: "registry export yaml" short description must refer to `the API Registry` instead of `the registry`.
=== RUN   TestCommandMetadata/registry_get
    root_test.go:73: "registry get" "filter" flag description must not be empty.
=== RUN   TestCommandMetadata/registry_label
    root_test.go:73: "registry label" "filter" flag description must not be empty.
    root_test.go:73: "registry label" "jobs" flag description must not be empty.
    root_test.go:73: "registry label" "overwrite" flag description must not be empty.
=== RUN   TestCommandMetadata/registry_resolve
    root_test.go:43: "registry resolve" short description must not begin with a lower case letter.
    root_test.go:46: "registry resolve" short description must refer to `the API Registry` instead of `the registry`.
    root_test.go:73: "registry resolve" "dry-run" flag description must not be empty.
=== RUN   TestCommandMetadata/registry_rpc
=== RUN   TestCommandMetadata/registry_upload
=== RUN   TestCommandMetadata/registry_upload_bulk
=== RUN   TestCommandMetadata/registry_upload_bulk_discovery
    root_test.go:53: "registry upload bulk discovery" has an empty 'Args' field.
    root_test.go:73: "registry upload bulk discovery" "service" flag description must not be empty.
=== RUN   TestCommandMetadata/registry_upload_bulk_openapi
    root_test.go:73: "registry upload bulk openapi" "base-uri" flag description must not be empty.
=== RUN   TestCommandMetadata/registry_upload_bulk_protos
    root_test.go:73: "registry upload bulk protos" "base-uri" flag description must not be empty.
    root_test.go:73: "registry upload bulk protos" "protoc-root" flag description must not be empty.
=== RUN   TestCommandMetadata/registry_upload_csv
    root_test.go:73: "registry upload csv" "delimiter" flag description must not be empty.
    root_test.go:73: "registry upload csv" "jobs" flag description must not be empty.
    root_test.go:73: "registry upload csv" "project-id" flag description must not be empty.
=== RUN   TestCommandMetadata/registry_upload_spec
    root_test.go:73: "registry upload spec" "style" flag description must not be empty.
    root_test.go:73: "registry upload spec" "version" flag description must not be empty.
=== RUN   TestCommandMetadata/registry_vocabulary
=== RUN   TestCommandMetadata/registry_vocabulary_difference
    root_test.go:73: "registry vocabulary difference" "output" flag description must not be empty.
=== RUN   TestCommandMetadata/registry_vocabulary_intersection
    root_test.go:73: "registry vocabulary intersection" "output" flag description must not be empty.
=== RUN   TestCommandMetadata/registry_vocabulary_union
    root_test.go:73: "registry vocabulary union" "output" flag description must not be empty.
=== RUN   TestCommandMetadata/registry_vocabulary_unique
    root_test.go:73: "registry vocabulary unique" "output-id" flag description must not be empty.
=== RUN   TestCommandMetadata/registry_vocabulary_versions
    root_test.go:73: "registry vocabulary versions" "output" flag description must not be empty.
--- PASS: TestCommandMetadata (0.00s)
    --- PASS: TestCommandMetadata/registry (0.00s)
    --- PASS: TestCommandMetadata/registry_annotate (0.00s)
    --- PASS: TestCommandMetadata/registry_apply (0.00s)
    --- PASS: TestCommandMetadata/registry_auth (0.00s)
    --- PASS: TestCommandMetadata/registry_auth_print-token (0.00s)
    --- PASS: TestCommandMetadata/registry_check (0.00s)
    --- PASS: TestCommandMetadata/registry_compute (0.00s)
    --- PASS: TestCommandMetadata/registry_compute_complexity (0.00s)
    --- PASS: TestCommandMetadata/registry_compute_conformance (0.00s)
    --- PASS: TestCommandMetadata/registry_compute_lint (0.00s)
    --- PASS: TestCommandMetadata/registry_compute_lintstats (0.00s)
    --- PASS: TestCommandMetadata/registry_compute_score (0.00s)
    --- PASS: TestCommandMetadata/registry_compute_scorecard (0.00s)
    --- PASS: TestCommandMetadata/registry_compute_vocabulary (0.00s)
    --- PASS: TestCommandMetadata/registry_config (0.00s)
    --- PASS: TestCommandMetadata/registry_config_configurations (0.00s)
    --- PASS: TestCommandMetadata/registry_config_configurations_activate (0.00s)
    --- PASS: TestCommandMetadata/registry_config_configurations_create (0.00s)
    --- PASS: TestCommandMetadata/registry_config_configurations_delete (0.00s)
    --- PASS: TestCommandMetadata/registry_config_configurations_describe (0.00s)
    --- PASS: TestCommandMetadata/registry_config_configurations_list (0.00s)
    --- PASS: TestCommandMetadata/registry_config_get (0.00s)
    --- PASS: TestCommandMetadata/registry_config_list (0.00s)
    --- PASS: TestCommandMetadata/registry_config_set (0.00s)
    --- PASS: TestCommandMetadata/registry_config_unset (0.00s)
    --- PASS: TestCommandMetadata/registry_count (0.00s)
    --- PASS: TestCommandMetadata/registry_count_revisions (0.00s)
    --- PASS: TestCommandMetadata/registry_count_versions (0.00s)
    --- PASS: TestCommandMetadata/registry_delete (0.00s)
    --- PASS: TestCommandMetadata/registry_diff (0.00s)
    --- PASS: TestCommandMetadata/registry_export (0.00s)
    --- PASS: TestCommandMetadata/registry_export_csv (0.00s)
    --- PASS: TestCommandMetadata/registry_export_yaml (0.00s)
    --- PASS: TestCommandMetadata/registry_get (0.00s)
    --- PASS: TestCommandMetadata/registry_label (0.00s)
    --- PASS: TestCommandMetadata/registry_resolve (0.00s)
    --- PASS: TestCommandMetadata/registry_rpc (0.00s)
    --- PASS: TestCommandMetadata/registry_upload (0.00s)
    --- PASS: TestCommandMetadata/registry_upload_bulk (0.00s)
    --- PASS: TestCommandMetadata/registry_upload_bulk_discovery (0.00s)
    --- PASS: TestCommandMetadata/registry_upload_bulk_openapi (0.00s)
    --- PASS: TestCommandMetadata/registry_upload_bulk_protos (0.00s)
    --- PASS: TestCommandMetadata/registry_upload_csv (0.00s)
    --- PASS: TestCommandMetadata/registry_upload_spec (0.00s)
    --- PASS: TestCommandMetadata/registry_vocabulary (0.00s)
    --- PASS: TestCommandMetadata/registry_vocabulary_difference (0.00s)
    --- PASS: TestCommandMetadata/registry_vocabulary_intersection (0.00s)
    --- PASS: TestCommandMetadata/registry_vocabulary_union (0.00s)
    --- PASS: TestCommandMetadata/registry_vocabulary_unique (0.00s)
    --- PASS: TestCommandMetadata/registry_vocabulary_versions (0.00s)
PASS
ok      github.com/apigee/registry/cmd/registry/cmd 
```

### CODE REVIEW

File: cmd/registry/cmd/get/get.go [hunk 1]
```diff
@@ -71,8 +71,8 @@ func Command() *cobra.Command {
 		},
 	}
 
-	cmd.Flags().StringVar(&filter, "filter", "", "Filter selected resources")
-	cmd.Flags().StringVarP(&output, "output", "o", "name", "Output type (name, yaml, contents)")
+	cmd.Flags().StringVar(&filter, "filter", "", "filter selected resources")
+	cmd.Flags().StringVarP(&output, "output", "o", "name", "output type (name, yaml, contents)")
```
Comment (theganyo) [inline cmd/registry/cmd/get/get.go:75]: Everywhere else we use an enum style of (name|yaml|contents). We should pick one.

File: cmd/registry/cmd/annotate/annotate.go [hunk 2]
```diff
@@ -81,9 +81,9 @@ func Command() *cobra.Command {
 		},
 	}
 
-	cmd.Flags().StringVar(&filter, "filter", "", "Filter selected resources")
-	cmd.Flags().BoolVar(&overwrite, "overwrite", false, "Overwrite existing annotations")
-	cmd.Flags().IntVar(&jobs, "jobs", 10, "Number of actions to perform concurrently")
+	cmd.Flags().StringVar(&filter, "filter", "", "filter selected resources")
+	cmd.Flags().BoolVar(&overwrite, "overwrite", false, "overwrite existing annotations")
+	cmd.Flags().IntVar(&jobs, "jobs", 10, "number of actions to perform concurrently")
```
Comment (theganyo) [inline cmd/registry/cmd/annotate/annotate.go:86]: We have various descriptions for `--jobs` that perhaps we should consolidate? Also, some commands include a `-j` short option and others don't.

**Review by theganyo — APPROVED** [review]
Nit: I made a couple of consistency notes we might want to go ahead and address now.

File: cmd/registry/cmd/get/get.go (hunk 1 above)
Comment (timburks) [reply cmd/registry/cmd/get/get.go:75]: Thanks, I'm adding a check that reports this:
```
    root_test.go:85: "registry get" "output" flag usage seems to contain an enum "(name, yaml, contents)", which should be of the form (a|b|c).
```

with a fix to `registry get`.

File: cmd/registry/cmd/annotate/annotate.go (hunk 2 above)
Comment (timburks) [reply cmd/registry/cmd/annotate/annotate.go:86]: Agreed: `jobs`, `filter`, `force` should probably all have consistent descriptions, etc.

Comment (timburks) [reply cmd/registry/cmd/annotate/annotate.go]: I've added a check that verifies that those three commands have consistent usage and shorthand options.

**Review by shrutiparabgoogle — APPROVED** [review]

**Pull request merged**